docs/src/configuration/app/source-operations.md

```diff
@@ -51,9 +51,109 @@ Note that these operations run in an isolated container which is not part of the
 
 Also, if multiple applications in a single project both result in a new commit, that will appear as two distinct commits in the Git history but only a single new build/deploy cycle will occur. If multiple applications define source operations with the same name, they will all be executed sequentially on each application.
 
+## Source Operations usage examples
+
+### Update a site from an upstream repository or template
+
+The following source operation syncronizes your branch with an upstream Git repository.
+
+1. Add a project-level variable named `env:UPSTREAM_REMOTE` with the Git URL of the upstream repository. That will make that repository available as a Unix environment variable in all environments, including in the Source Operations environment.
+
+- Variable name: `env:UPSTREAM:REMOTE`
+- Variable example value: `https://github.com/platformsh/platformsh-docs`
+
+2. In your  `.platform.app.yaml` file, define a Source Operation to fetch from that upstream repository:
+
+```
+source:
+    operations:
+        upstream-update:
+            command: |
+                set -e
+                git remote add upstream $UPSTREAM_REMOTE
+                git fetch --all
+                git merge upstream/master
+```
+
+
+Now every time you run `platform source-operation:run upstream-update` using the CLI on a given branch, the branch fetches all changes from the upstream git repository and then merges the latest changes from the `master` branch in the upstream repository.
+If there’s a conflict merging from the upstream repository, the source operation will fail and not update from upstream.
+
+Run the `upstream-update` operation on a Development environment rather than directly on Production.
+
+### Revert to the last commit
+
+The following source operation will revert the last commit pushed to the Git repository. This can be useful if you did not properly test the changes of another operation, and you need to quickly revert to the previous state.
```

Comment:
```suggestion
The following Source Operation will revert the last commit pushed to the Git repository. This can be useful if you did not properly test the changes of another operation, and you need to quickly revert to the previous state.
```